audit-info/src/error.rs

```diff
@@ -0,0 +1,72 @@
+#[derive(Debug)]
+pub enum Error {
```

Comment:
nit: I generally hate calling them Error and try to be specific e.g. AuditableInfoError or smth - may be convoluted

You could also replace whole lot of manual error handlign with `thiserror` and make it more compact

Reply:
`AuditableInfoError` is no better than `auditable_info::Error`, so I'll keep it that way.

`thiserror` is deliberately avoided here because it introduces unsafe code, and this is going into `rust-audit-info` binary which has a 100% safe code policy. I also wanted to avoid supply chain attack surface. So yes, I understand the boilerplate could be avoided, but it involves trade-offs and in this particular case the boilerplate is a deliberate decision.

Reply:
Problem with Error is that I may have conflicting Error type and I always get lost between Error types when having to deal with a lot of different errors coming from many places.

If I read it correctly I would have to use the `crate::Error as SomethingElseError` and then everyone invents their own `crate::Error as WhateverError` ?

Reply:
When multiple types are named simply `Error`, the compiler enforces that you address one as `Error` and the other as `some_module::Error`, for example:

https://github.com/rustsec/rustsec/blob/a104cfb4d38e0d47295546ee13f5cf7bd3de315c/rustsec/src/error.rs#L116

Since the non-native one is always addressed as `some_module::Error`, there should be no confusion. Also, notice how many errors converted in there are simply `Error` in their module.